src/cascadia/TerminalControl/UiaTextRange.cpp

```diff
@@ -0,0 +1,194 @@
+{
+    try
+    {
+        return new UiaTextRange(pData, pProvider, point);
+    }
+    catch (...)
+    {
+        return nullptr;
+    }
+}
+
+// degenerate range constructor.
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData, _In_ IRawElementProviderSimple* const pProvider) :
+    UiaTextRangeBase(pData, pProvider)
+{
+}
+
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData,
+                           _In_ IRawElementProviderSimple* const pProvider,
+                           const Cursor& cursor) :
+    UiaTextRangeBase(pData, pProvider, cursor)
+{
+}
+
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData,
+                           _In_ IRawElementProviderSimple* const pProvider,
+                           const Endpoint start,
+                           const Endpoint end,
+                           const bool degenerate) :
+    UiaTextRangeBase(pData, pProvider, start, end, degenerate)
+{
+}
+
+// returns a degenerate text range of the start of the row closest to the y value of point
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData,
+                           _In_ IRawElementProviderSimple* const pProvider,
+                           const UiaPoint point) :
+    UiaTextRangeBase(pData, pProvider)
+{
+    Initialize(point);
+}
+
+IFACEMETHODIMP UiaTextRange::Clone(_Outptr_result_maybenull_ ITextRangeProvider** ppRetVal)
+{
+    try
+    {
+        *ppRetVal = new UiaTextRange(*this);
+    }
+    catch (...)
+    {
+        *ppRetVal = nullptr;
+        return wil::ResultFromCaughtException();
+    }
+    if (*ppRetVal == nullptr)
+    {
+        return E_OUTOFMEMORY;
+    }
+
+#if defined(_DEBUG) && defined(UiaTextRangeBase_DEBUG_MSGS)
+    OutputDebugString(L"Clone\n");
+    std::wstringstream ss;
+    ss << _id << L" cloned to " << (static_cast<UiaTextRangeBase*>(*ppRetVal))->_id;
+    std::wstring str = ss.str();
+    OutputDebugString(str.c_str());
+    OutputDebugString(L"\n");
+#endif
+    // TODO GitHub #1914: Re-attach Tracing to UIA Tree
+    // tracing
+    /*ApiMsgClone apiMsg;
+    apiMsg.CloneId = static_cast<UiaTextRangeBase*>(*ppRetVal)->GetId();
+    Tracing::s_TraceUia(this, ApiCall::Clone, &apiMsg);*/
+
+    return S_OK;
+}
+
+IFACEMETHODIMP UiaTextRange::FindText(_In_ BSTR text,
+                                      _In_ BOOL searchBackward,
+                                      _In_ BOOL ignoreCase,
+                                      _Outptr_result_maybenull_ ITextRangeProvider** ppRetVal)
+{
+    // TODO GitHub #605: Search functionality
+    return E_NOTIMPL;
+}
+
+void UiaTextRange::_ChangeViewport(const SMALL_RECT NewWindow)
+{
+    // TODO GitHub #2361: Update viewport when calling UiaTextRangeBase::ScrollIntoView()
+}
+
+void UiaTextRange::_TranslatePointToScreen(LPPOINT clientPoint) const
+{
+    // TODO GitHub #2103: NON-HWND IMPLEMENTATION OF CLIENTTOSCREEN()

```

Comment:
Why is this not crying about UNREFERENCED_PARAMETER? Are warnings off? (and the above/below)